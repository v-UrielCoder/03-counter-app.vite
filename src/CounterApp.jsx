import { useState } from 'react'
import PropTypes from 'prop-types'



export const CounterApp = ({ value }) => {
    
    const [ contador, setCounter  ] = useState(value)

    const handleAdd = () => {
        setCounter( contador + 1)
    }
    const decrement = () => {
        setCounter( (c) => c - 1);
    }
    const reset = () => {
        setCounter(value);
    }
    
    
    return(
        <>
            <h1>CounterApp</h1>
            <h2> { contador } </h2>

            <button onClick={handleAdd}>+1</button>
            <button onClick={decrement}>-1</button>
            <button  aria-label='btn-reset' onClick={reset}>Reset</button>
        </>
        
    )
}  

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}