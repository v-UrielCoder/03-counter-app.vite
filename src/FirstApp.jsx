import PropTypes from 'prop-types'


export const FirstApp = ({ title, subTitle, name }) => {
    
    
    return (
        <>
            <h1 data-testid = 'test-title'>   {title}  </h1>
            <p data-testid = 'test-name'>{name}</p>
            <p>{subTitle} </p>
            <p>{subTitle} </p>
            <p>{subTitle} </p>
            
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle : PropTypes.string
}

FirstApp.defaultProps = {
    name: 'Uriel Emiliano',
    subTitle : "No hay subtitulo"
}
    
    
    
