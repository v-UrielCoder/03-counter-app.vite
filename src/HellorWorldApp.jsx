
export const HellorWorldApp = () => {
  return (
    <div>HellorWorldApp</div>
  )
}


