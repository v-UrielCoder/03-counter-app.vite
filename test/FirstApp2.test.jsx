import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en FirstApp', () => {

    const title = 'Hola, Soy Goku';
    const subTitle = 'No hay subtitulo';

    test('debe de hacer match con el snapshot', () => {

        const { container } = render( <FirstApp title={ title }/>);
        expect( container ).toMatchSnapshot(); 
    })

    test('debe de mostart el mensaje "Hola, Soy Goku"', () => {

        render( <FirstApp title= { title } /> );
        expect( screen.getByText(title)).toBeTruthy();
    })
    test('debe de mostart el titulo en un h1', () => {
        
        render( <FirstApp title= { title } /> );
        expect( screen.getByRole('heading' , { level: 1 }).innerHTML ).toContain(title)
    })
    test('Debe de mostrar el subtitulo mostrado por props', () => {

        render( <FirstApp
            title = {title}
            subTitle = {subTitle}
            />)
        
            expect( screen.getAllByText(subTitle).length).toBe(3)
    })
})