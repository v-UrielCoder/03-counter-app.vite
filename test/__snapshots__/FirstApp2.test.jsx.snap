// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
       
    Hola, Soy Goku
      
  </h1>
  <p
    data-testid="test-name"
  >
    Uriel Emiliano
  </p>
  <p>
    No hay subtitulo
     
  </p>
  <p>
    No hay subtitulo
     
  </p>
  <p>
    No hay subtitulo
     
  </p>
</div>
`;
